### PR TITLE
[jasmine-ajax] Add missing response properties: statusText, responseURL, responseJSON

### PR DESCRIPTION
**Pull request by Finesse**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test YOUR_PACKAGE_NAME`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/jasmine/jasmine-ajax/blob/faed4ca10297b1c8d4f0ba13491067f34a1d88ab/src/fakeRequest.js#L257
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/jasmine-ajax/index.d.ts
Comment (Finesse) [inline types/jasmine-ajax/index.d.ts:11]: I'd like to remove this type in favor of `JasmineAjaxRequestStubReturnOptions`, but it will be a breaking change, so I've left it and marked it as deprecated.

File: types/jasmine-ajax/index.d.ts
Comment (jgonggrijp) [inline types/jasmine-ajax/index.d.ts:15]: Is `JasmineAjaxResponse` really *necessarily* the same type as `JasmineAjaxRequestStubReturnOptions`? I mean, I know it *currently* is the same, but is it completely out of the question that Jasmine will some day add an option to `stubRequest.andReturn()` that isn't directly reflected in the fake response?

File: types/jasmine-ajax/index.d.ts
Comment (jgonggrijp) [inline types/jasmine-ajax/index.d.ts:11]: If the answer to my previous question was "yes", then I'd much prefer that you deprecate `JasmineAjaxRequestStubReturnOptions` rather than `JasmineAjaxResponse`.

**Review by jgonggrijp — CHANGES_REQUESTED** [review]
@Finesse Thanks for going out of your way to fix this.

File: types/jasmine-ajax/index.d.ts
Comment (Finesse) [reply types/jasmine-ajax/index.d.ts:15]: I tried to make the `andReturn` code look like the `andError` code. Both `andError` and `responseError` use a single argument type: `JasmineAjaxRequestStubErrorOptions`.

Actually it's incorrect to use the same type both in the error methods and in the success methods because the `status` field is required both in `respondWith` and `responseError`. But since the original library author violates this requirement in the tests, I've decided to leave it as is.

Comment (Finesse) [reply types/jasmine-ajax/index.d.ts:15]: Also there is no response interface in XMLHttpRequest, the response data is stored in the `XMLHttpRequest` object directly. So I believe that having a type called `JasmineAjaxResponse` is a mistake.

But if you don't agree, I'll just make as you ask.

Comment (jgonggrijp) [reply types/jasmine-ajax/index.d.ts:15]: I think I'm misunderstanding your arguments, because I don't see how they are related to my question. Has the content of the response object returned by a stub request necessarily the same type as the information passed to `.andReturn`, yes or no?

Comment (Finesse) [reply types/jasmine-ajax/index.d.ts:15]: Your question looks incorrect because there is no function that returns a _response_ object in the type definition. I guess you want to know whether the `JasmineAjaxRequest.respondWith` and `JasmineAjaxRequestStub.andReturn` function argument types are the same type or not. If so, then the answer is "no" because the function are independent. But at the moment the types are equal.

Considering this and the way the `JasmineAjaxRequestStubErrorOptions` type is used, I've decided to do this MR the way it's done.

Comment (jgonggrijp) [reply types/jasmine-ajax/index.d.ts:15]: Ah, now I see what you mean. Yes, I agree that my question about returning a response object was a bit misguided. I think I assumed that such a thing was going on because of the name `JasmineAjaxResponse`. I now understand it's just the argument type of `respondWith`. Should have checked the code better before asking my question, sorry about that.

Anyway, [the documentation](https://jasmine.github.io/tutorials/mocking_ajax) seems to suggest these types are actually intended to be the same, so you were right to merge them:

> Call `stubRequest` with the url you want to return immediately. Then `andReturn` receives the same type of argument as `respondWith`.

File: types/jasmine-ajax/index.d.ts
Comment (jgonggrijp) [reply types/jasmine-ajax/index.d.ts:11]: I still prefer this, simply because the latter name is shorter and the former is too specific (the type is not only for `stubRequest.andReturn` but also for `respondWith`.

Comment (Finesse) [reply types/jasmine-ajax/index.d.ts:11]: Ok, I'll make this solution:

> I'd much prefer that you deprecate `JasmineAjaxRequestStubReturnOptions` rather than `JasmineAjaxResponse`.

**Review by jgonggrijp — APPROVED** [review]

File: types/jasmine-ajax/index.d.ts
Comment (jgonggrijp) [reply types/jasmine-ajax/index.d.ts]: Thanks!

**Comment by Finesse** [comment]
Ready to merge

**Comment by Finesse** [comment]
@jgonggrijp Probably you should approve the MR once again in order for the bot to merge it

**Review by jgonggrijp — APPROVED** [review]

**Comment by jgonggrijp** [comment]
Ready to merge

**Pull request merged**
